src/components/text-input/index.md.njk

```diff
@@ -21,9 +21,15 @@ Do not use the text input component if you need to let users enter longer answer
 
 ## How it works
 
-All text inputs must have visible labels; placeholder text is not an acceptable replacement for a label as it vanishes when users start typing.
+Text inputs should usually have visible labels. However, in some cases, it's okay to use a visually hidden label. For example, if you're progressively revealing a text input with radios or checkboxes, and the input would otherwise be the same as the parent label.
```

Comment:
Looking good!

The 2nd sentence in this paragraph feels a bit clunky... Is this talking about conditionally revealed question?

If so, it'd be good to [match the wording we use in the Radios and Checkboxes components](https://design-system.service.gov.uk/components/radios/#conditionally-revealing-a-related-question).

Also, does this use case match the example of the page? If so we might need to explain why they're not visually hidden there.

Reply:
Cheers for reviewing! Aye, it's about conditionally revealed questions - although I now think we're probably not going to go into specifics about edge cases. (See David's suggestion below.)